jvcl/run/JvMemoryDataset.pas

```diff
@@ -833,7 +834,13 @@ procedure TJvMemoryData.InitFieldDefsFromFields;
     begin
       FOffsets[I] := Offset;
       if FieldDefList[I].DataType in ftSupported - ftBlobTypes then
-        Inc(Offset, CalcFieldLen(FieldDefList[I].DataType, FieldDefList[I].Size) + 1);
+      begin
+        FieldLen := CalcFieldLen(FieldDefList[I].DataType, FieldDefList[I].Size);
+        if Offset + FieldLen + 1 <= high(Offset) then
+          Inc(Offset, FieldLen + 1)
+        else
+          raise EIntOverflow.CreateResFmt(RsEFieldOffsetOverflow, [I]);
```

Comment:
To me, `EIntOverflow` is when an `Integer` value goes over `MaxInt` and here, I believe it's more an "out of range" issue, so `ERangeError` seems better to me

Reply:
Ok, added another commit where I changed the exception thrown. I hope it's good for the merge now ;-) When implementing the first variant I really had both exception classes in mind...